dbms/src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -974,9 +974,19 @@ bool ReplicatedMergeTreeQueue::shouldExecuteLogEntry(
           * because the leader replica does not assign merges of greater size (except OPTIMIZE PARTITION and OPTIMIZE FINAL).
           */
         const auto data_settings = data.getSettings();
-        bool ignore_max_size = (entry.type == LogEntry::MERGE_PARTS) && (max_source_parts_size == data_settings->max_bytes_to_merge_at_max_space_in_pool);
+        bool ignore_max_size_for_merge  = (entry.type == LogEntry::MERGE_PARTS) && (max_source_parts_size == data_settings->max_bytes_to_merge_at_max_space_in_pool);
+        bool ignore_max_size_for_mutate = (entry.type == LogEntry::MUTATE_PART) && (max_source_parts_size == data_settings->max_bytes_to_mutate_at_max_space_in_pool);
```

Comment:
Please take a look at `MergeTreeDataMergerMutator::getMaxSourcePartSizeForMutation()`. It calculates `max_source_parts_size` as `disk_space / DISK_USAGE_COEFFICIENT_TO_RESERVE`, so it almost never equals to `max_bytes_to_mutate_at_max_space_in_pool` and `ignore_max_size_for_mutate` is false